src/backend/config/settings.py

```diff
@@ -0,0 +1,264 @@
+    gdrive: Optional[GDriveSettings]
+
+
+class DatabaseSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    url: Optional[str] = Field(validation_alias=AliasChoices("DATABASE_URL", "url"))
+    migrate_token: Optional[str] = Field(
+        validation_alias=AliasChoices("MIGRATE_TOKEN", "migrate_token")
+    )
+
+
+class SageMakerSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    endpoint_name: Optional[str] = Field(
+        validation_alias=AliasChoices("SAGE_MAKER_ENDPOINT_NAME", "endpoint_name")
+    )
+    region_name: Optional[str] = Field(
+        validation_alias=AliasChoices("SAGE_MAKER_REGION_NAME", "region_name")
+    )
+    access_key: Optional[str] = Field(
+        validation_alias=AliasChoices("SAGE_MAKER_ACCESS_KEY", "access_key")
+    )
+    secret_key: Optional[str] = Field(
+        validation_alias=AliasChoices("SAGE_MAKER_SECRET_KEY", "secret_key")
+    )
+    session_token: Optional[str] = Field(
+        validation_alias=AliasChoices("SAGE_MAKER_SESSION_TOKEN", "session_token")
+    )
+
+
+class AzureSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    endpoint_url: Optional[str] = Field(
+        validation_alias=AliasChoices("AZURE_CHAT_ENDPOINT_URL", "endpoint_url")
+    )
+    api_key: Optional[str] = Field(
+        validation_alias=AliasChoices("AZURE_API_KEY", "api_key")
+    )
+
+
+class CoherePlatformSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    api_key: Optional[str] = Field(
+        validation_alias=AliasChoices("COHERE_API_KEY", "api_key")
+    )
+
+
+class SingleContainerSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    model: Optional[str] = Field(
+        validation_alias=AliasChoices("SINGLE_CONTAINER_MODEL", "model")
+    )
+    url: Optional[str] = Field(
+        validation_alias=AliasChoices("SINGLE_CONTAINER_URL", "url")
+    )
+
+
+class BedrockSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    region_name: Optional[str] = Field(
+        validation_alias=AliasChoices("BEDROCK_REGION_NAME", "region_name")
+    )
+    access_key: Optional[str] = Field(
+        validation_alias=AliasChoices("BEDROCK_ACCESS_KEY", "access_key")
+    )
+    secret_key: Optional[str] = Field(
+        validation_alias=AliasChoices("BEDROCK_SECRET_KEY", "secret_key")
+    )
+    session_token: Optional[str] = Field(
+        validation_alias=AliasChoices("BEDROCK_SESSION_TOKEN", "session_token")
+    )
+
+
+class DeploymentSettings(BaseSettings, BaseModel):
+    model_config = setting_config
+    default_deployment: Optional[str]
+    enabled_deployments: Optional[List[str]]
+
+    sagemaker: Optional[SageMakerSettings]
+    azure: Optional[AzureSettings]
+    cohere_platform: Optional[CoherePlatformSettings]
+    single_container: Optional[SingleContainerSettings]
+    bedrock: Optional[BedrockSettings]
+
+
+class Settings(BaseSettings, case_sensitive=False):
```

Comment:
Should we initialize a settings instance once and put it in `backend/config/__init__.py` with @lru_cache something like
```
@lru_cache(maxsize=1)
def settings() -> Settings:
    try:
        return Settings()
    except Exception as e:
        # handle
```

Prevents from initializing many Settings instances

Reply:
Added, is the error handling necessary? I find the errors from settings quite good